tests/lib/rules/display-name.js

```diff
@@ -672,6 +688,94 @@ ruleTester.run('display-name', rule, {
     errors: [{
       message: 'Component definition is missing display name'
     }]
+  }, {
+    code: `
+      import React from 'react'
+
+      export const ComponentWithMemo = React.memo(({ world }) => {
+        return <div>Hello {world}</div>
+      })
+    `,
+    parser: parsers.BABEL_ESLINT,
+    errors: [{
+      message: 'Component definition is missing display name'
+    }]
+  }, {
+    code: `
+      import React from 'react'
+
+      export const ComponentWithMemo = React.memo(function() {
+        return <div>Hello {world}</div>
+      })
+    `,
+    parser: parsers.BABEL_ESLINT,
+    errors: [{
+      message: 'Component definition is missing display name'
+    }]
+  }, {
+    code: `
+      import React from 'react'
+
+      export const ComponentWithForwardRef = React.forwardRef(({ world }) => {
+        return <div>Hello {world}</div>
+      })
+    `,
+    parser: parsers.BABEL_ESLINT,
+    errors: [{
+      message: 'Component definition is missing display name'
+    }]
+  }, {
+    code: `
+      import React from 'react'
+
+      export const ComponentWithForwardRef = React.forwardRef(function({ world }) {
+        return <div>Hello {world}</div>
+      })
+    `,
+    parser: parsers.BABEL_ESLINT,
+    errors: [{
+      message: 'Component definition is missing display name'
+    }]
+  }, {
+    // This test shows buggy behaviour but the outcome for the user is kinda
+    // right if you squint at it
+    // In this test the plugin identifies two components instead of one.
+    // React is unable to work out the displayName for this code so we're
+    // reporting the right thing, just doing it twice.
+    // I think ideally this should only report one error.
+    code: `
+      import React from 'react'
+
+      export const ComponentWithMemoAndForwardRef = React.memo(
+        React.forwardRef(({ world }) => {
+          return <div>Hello {world}</div>
+        })
+      )
+    `,
+    parser: parsers.BABEL_ESLINT,
+    errors: [{
+      message: 'Component definition is missing display name', line: 4
+    }, {
+      message: 'Component definition is missing display name', line: 5
+    }]
+  }, {
+    // React does not currently set a displayName on the memo function when you
+    // pass it a value from forwardRef.
+    // Hopefully eventually that will be fixed, but until then flagging this
```

Comment:
this shouldn't be fixed because it's not broken; forwardRef is a component but does not accept one; it accepts a callback that takes `props` and `ref`.

Reply:
I think I'm referring to a slightly different issue. I'm referring how React.memo handles accepting a Component vs the result of calling forwardRef.

When inspecting the MemoFunction and  MemoThenForwardRefFunction 
 items on https://codesandbox.io/s/awesome-paper-hh0wh?module=App.js I end up with the following in my react devtools:

<img width="496" alt="Screen Shot 2019-09-02 at 11 28 43 PM" src="https://user-images.githubusercontent.com/227292/64148938-753c6380-cdd9-11e9-8e8c-7f8ecaa72dad.png">

Note that calling React.memo with a function component defined inline

```jsx
const MemoFunction = React.memo(function MemoFunction() {
  return <div>MemoFunction</div>;
});
```

Results in in the name `MemoFunction [memo]` in the component tree.

However calling React.memo with an invocation of React.forwardRef (which returns a Component)

```jsx
const MemoThenForwardRefFunction = React.memo(
  React.forwardRef(function MemoThenForwardRefFunction() {
    return <div>MemoThenForwardRefFunction</div>;
  })
);
```

Results in the name `Anonymous` in the component tree  - Given that the result of an unwrapped call to React.forwardRef has a meaningful name, I would have expected that to have been picked up by React.memo

Reply:
I've raised this upstream as https://github.com/facebook/react/issues/16722